src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -143,3 +159,198 @@ const filterDecorators = (
+ * ```
+ *
+ * This plays nicely with later using `Object.defineProperty` on the prototype
+ * to define getters and setters, or simply with defining them right on the
+ * class (see the code in `proxyComponent`, `proxyCustomElement`, and friends).
+ *
+ * However, with a `target` of ES2022 or higher (e.g. `ESNext`) default
+ * behavior for TypeScript is instead to emit code like this:
+ *
+ * ```ts
+ * class CompiledMyClass {
+ *   foo = "bar"
+ * }
+ * ```
+ *
+ * This output is more correct because the compiled code 1) more closely
+ * resembles the TypeScript source and 2) is using standard JS syntax instead
+ * of desugaring it. There is an announcement in the release notes for
+ * TypeScript v3.7 which explains some helpful background about the change,
+ * and about the `useDefineForClassFields` TypeScript option which lets you
+ * opt-in to the old output:
+ *
+ * https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier
+ *
+ * For our use-case, however, the ES2022+ behavior doesn't work, since we need
+ * to be able to define getters and setters on these fields. We could require
+ * that the TypeScript configuration used for Stencil have the
+ * `useDefineForClassFields` setting set to `false`, but that would have the
+ * undesirable side-effect that class fields which are _not_
+ * decorated with a Stencil decorator would also be instantiated in the
+ * constructor.
+ *
+ * So instead, we take matters into our own hands. When we encounter a class
+ * field which is decorated with a Stencil decorator we remove it from the
+ * class and add a statement to the constructor to instantiate it with the
+ * correct default value.
+ *
+ * **Note**: this function will modify a constructor if one is already present on
+ * the class or define a new one otherwise.
+ *
+ * @param classNode a TypeScript AST node for a Stencil component class
+ * @param classMembers the class members that we need to update
+ * @returns a list of updated class elements which can be inserted into the class
+ */
+function handleClassFields(classNode: ts.ClassDeclaration, classMembers: ts.ClassElement[]): ts.ClassElement[] {
+  const statements = [];
+  const updatedClassMembers = [];
+
+  for (const member of classMembers) {
+    const currentDecorators = member.decorators;
+    const newDecorators = filterDecorators(member, MEMBER_DECORATORS_TO_REMOVE);
+    if (currentDecorators !== newDecorators && ts.isPropertyDeclaration(member)) {
+      // this is a class field that we'll need to handle, so lets push a statement for
+      // initializing the value onto our statements list
+      statements.push(
+        ts.factory.createExpressionStatement(
+          ts.factory.createBinaryExpression(
+            ts.factory.createPropertyAccessExpression(
+              ts.factory.createThis(),
+              // @ts-ignore
+              ts.factory.createIdentifier(member.name.escapedText)
```

Comment:
Can we determine if there's another property that doesn't require us to use `@ts-ignore`? I've been bit by that one being undefined before (or maybe it was `text`?), IDK if that's what we want to move forward with or not

Reply:
Came up with this, it's a bit wordy but no type assertions or `@ts-ignore` necessary:

```ts
      const declarationName: ts.DeclarationName = ts.getNameOfDeclaration(member);

      const memberName =
        declarationName.kind === ts.SyntaxKind.ComputedPropertyName
          ? declarationName.expression.getText()
          : declarationName.getText();
```

Reply:
`memberName` is then a string that we can just pass in to `ts.factory.createIdentifier`